examples/CosaMQTTsecureclient/CosaMQTTsecureclient.ino

```diff
@@ -0,0 +1,225 @@
+static const uint8_t flag = MQTT::Client::WILL_FLAG |                   // Connection flags; combine flag for will, username and password
+                            MQTT::Client::USER_NAME_FLAG |
+                            MQTT::Client::PASSWORD_FLAG;
+
+/*
+ * MQTT client; Parses incoming topic/message
+ */
+class MQTTClient : public MQTT::Client {
+public:
+  virtual void on_publish(char* topic, void* buf, size_t count);
+};
+
+void
+MQTTClient::on_publish(char* topic, void* buf, size_t count)
+{
+  String toparse((char*) topic);
+  String partition;
+  String payload((char*) buf);
+  
+  #if defined(DEVMODE)
+  // Print the topic and value to trace stream
+  trace << PSTR("on_publish::count = ") << count << endl;
+  trace << PSTR("  topic = ") << topic << endl;
+  trace << PSTR("  payload = ") << payload << endl;
+  #endif
+  
+  // Parse NEXA Switches
+  String key(toparse.substring(0, 24));
+  if (key.equals("nexa/switch/"))
+  {
+    String val(toparse.substring(24, toparse.length()));
+    uint8_t len = val.length();
+    uint16_t device = 0;
+    for(int i=0; i<len; i++){
+      device = device * 10 + ( val[i] - '0' );
+    }
+    
+    if (payload.equals("ON") || payload.equals("1"))
+    {
+      trace << PSTR("  NEXA Switch ") << device << PSTR(" turned ON") << endl;
+    }
+    else if(payload.equals("OFF") || payload.equals("0"))
+    {
+      trace << PSTR("  NEXA Switch ") << device << PSTR(" turned OFF") << endl;
+    }
+    
+    return;
+  } // end parsing NEXA Switches
+  
+  // Parse NEXA Groups
+  key = toparse.substring(0, 23);
+  if (key.equals("nexa/group/"))
+  {
+    String val(toparse.substring(23, toparse.length()));
+    uint8_t len = val.length();
+    uint16_t group = 0;
+    for(int i=0; i<len; i++){
+      group = group * 10 + ( val[i] - '0' );
+    }
+    
+    if (payload.equals("ON") || payload.equals("1"))
+    {
+      trace << PSTR("  NEXA Group ") << group << PSTR(" turned ON") << endl;
+    }
+    else if(payload.equals("OFF") || payload.equals("0"))
+    {
+      trace << PSTR("  NEXA Group ") << group << PSTR(" turned OFF") << endl;
+    }
+    
+    return;
+  } // end parsing NEXA Groups
+}
+
+// MQTT client
+MQTTClient client;
+
+void setup() {
+  // The program start when the serial debug terminal is connected
+  uart.begin(9600);
+  trace.begin(&uart, PSTR("Cosa MQTT secure client: started"));
+  Watchdog::begin();
+  
+  // Start the Ethernet Controller and request network address for hostname
+  ASSERT(ethernet.begin_P(CLIENT));
+
+  // Start the MQTT client
+  mqttStart();
+}
+
+void loop()
+{
+  // Service incoming publish messages until error
+  trace << PSTR("Service incomming messages") << endl;
+
+  int service = 0;
+  while (service > -1 )
+  {
+    TRACE(service = client.service());
+  }
+  
+  // Restart the client
+  trace << PSTR("Cosa-MQTT secure client: restarting") << endl;
+  mqttStop();
```

Comment:
Yes, good idea. I have not had any problems with this code though, but I run my broker locally. Might be different with a remote broker.